Upstream-Status: Inappropriate [OpenXT; old pesign SRCREV vs efivar 39]

pesign's efitypes.h and efivar's efivar-types.h both used the EFI_TYPES_H
include guard. Defining it in pesign first caused efivar to skip efi_guid_t.

--- a/src/efitypes.h	2026-08-18 04:01:57.868034247 -0400
+++ b/src/efitypes.h	2026-08-18 04:02:18.764034120 -0400
@@ -16,27 +16,16 @@
  *
  * Author(s): Peter Jones <pjones@redhat.com>
  */
-#ifndef EFI_TYPES_H
-#define EFI_TYPES_H 1
+#ifndef PESIGN_EFI_TYPES_H
+#define PESIGN_EFI_TYPES_H 1
 
+/* Include efivar first. Do not use EFI_TYPES_H as the include guard: efivar's
+ * efivar-types.h also uses that name, and defining it here first causes
+ * efi_guid_t / efi_time_t to be skipped.
+ */
 #include <efivar.h>
 
-typedef unsigned long efi_status_t;
-typedef uint16_t efi_char16_t;
-
-typedef struct {
-	uint16_t	year;
-	uint8_t		month;
-	uint8_t		day;
-	uint8_t		hour;
-	uint8_t		minute;
-	uint8_t		second;
-	uint8_t		pad1;
-	uint32_t	nanosecond;
-	int16_t		timezone;
-	uint8_t		daylight;
-	uint8_t		pad2;
-} efi_time_t;
+/* efi_status_t, efi_char16_t, efi_time_t, efi_guid_t come from efivar. */
 
 struct efi_variable {
 	efi_char16_t *VariableName;
@@ -47,17 +36,21 @@
 	uint32_t Attributes;
 };
 
-#define EFI_VARIABLE_NON_VOLATILE	0x0000000000000001
-#define EFI_VARIABLE_BOOTSERVICE_ACCESS	0x0000000000000002
-#define EFI_VARIABLE_RUNTIME_ACCESS	0x0000000000000004
-#define EFI_VARIABLE_HARDWARE_ERROR_RECORD	0x0000000000000008
-#define EFI_VARIABLE_AUTHENTICATED_WRITE_ACCESS	0x0000000000000010
-#define EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS 0x0000000000000020
-#define EFI_VARIABLE_APPEND_WRITE	0x0000000000000040
-
-#define EFI_GLOBAL_PLATFORM_KEY L"PK"
-#define EFI_GLOBAL_KEY_EXCHANGE_KEY L"KEK"
-#define EFI_IMAGE_SECURITY_DATABASE L"db"
-#define EFI_IMAGE_SECURITY_DATABASE1 L"dbx"
+#ifndef EFI_VARIABLE_NON_VOLATILE
+#define EFI_VARIABLE_NON_VOLATILE				0x0000000000000001
+#define EFI_VARIABLE_BOOTSERVICE_ACCESS				0x0000000000000002
+#define EFI_VARIABLE_RUNTIME_ACCESS				0x0000000000000004
+#define EFI_VARIABLE_HARDWARE_ERROR_RECORD			0x0000000000000008
+#define EFI_VARIABLE_AUTHENTICATED_WRITE_ACCESS			0x0000000000000010
+#define EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS	0x0000000000000020
+#define EFI_VARIABLE_APPEND_WRITE				0x0000000000000040
+#endif
+
+#ifndef EFI_GLOBAL_PLATFORM_KEY
+#define EFI_GLOBAL_PLATFORM_KEY		L"PK"
+#define EFI_GLOBAL_KEY_EXCHANGE_KEY	L"KEK"
+#define EFI_IMAGE_SECURITY_DATABASE	L"db"
+#define EFI_IMAGE_SECURITY_DATABASE1	L"dbx"
+#endif
 
-#endif /* EFI_TYPES_H */
+#endif /* PESIGN_EFI_TYPES_H */
